contracts/libraries/OracleLibrary.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0 <0.8.0;
+
+import '@uniswap/v3-core/contracts/libraries/FullMath.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/LowGasSafeMath.sol';
+import '../libraries/PoolAddress.sol';
+
+/// @title Oracle library
+/// @notice Provides functions to integrate with V3 pool oracle
+library OracleLibrary {
+    /// @notice Fetches time-weighted average tick using Uniswap V3 oracle
+    /// @param factory Address of Uniswap V3 factory contract
+    /// @param token0 Address of an ERC20 token contract
+    /// @param token1 Address of an ERC20 token contract
+    /// @param fee Fee amount that correspond to the pool that we want to observe
+    /// @param period Number of seconds in the past to start calculating time-weighted average
+    /// @return timeWeightedAverageTick The time-weighted average tick from (block.timestamp - period) to block.timestamp
+    function consult(
+        address factory,
```

Comment:
wait.. the reason we changed this was because we had to query token0, token1, but now it just runs the twap tick, so you can just take the pool

Reply:
it can also greatly simplify testing if you just write a mock pool that only implements `#observe`